libraries/libraries_api_test.go

```diff
@@ -114,6 +123,12 @@ func TestWaitForLibrariesInstalled(t *testing.T) {
 			"failed-wheel", 50 * time.Millisecond, true, true,
 		})
 		assert.NoError(t, err, "library should have been uninstalled and work proceeded")
+
+		// Cluster not available or doesn't exist
+		_, err = libs.WaitForLibrariesInstalled(Wait{
+			"1005-abcd", 50 * time.Millisecond, false, false,
+		})
+		assert.EqualError(t, err, "Cluster 1005-abcd does not exist")
```

Comment:
can you also check for 404 StatusCode of APIError here? this is the condition that triggers resource deletion.